test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointTrimTest.java

```diff
@@ -210,21 +210,76 @@ public void rawStreamConsumer() {
                 .setStreamName(tableName)
                 .open();
 
-        IStreamView s = newRuntime.getStreamsView().get(CorfuRuntime.getStreamID(tableName));
-        s.seek(BATCH_SIZE + CHECKPOINT_SIZE);
-        // Seek beyond the last trimmed address.
+        // Create a new stream and seek beyond the last trimmed address.
+        IStreamView stream = newRuntime.getStreamsView().get(CorfuRuntime.getStreamID(tableName));
+        stream.seek(BATCH_SIZE + CHECKPOINT_SIZE);
+
         // The first call to remainingUpTo() will load the checkpoint, and the
         // second one will fetch the actual data.
-        Assertions.assertThat(Stream.of(s.remainingUpTo(Long.MAX_VALUE), s.remainingUpTo(Long.MAX_VALUE))
+        Assertions.assertThat(Stream.of(
```

Comment:
Getting rid of `return true;` statement in `if (context.getCheckpoint() == StreamCheckpoint.UNINITIALIZED) ` block and replacing other returns with `return areReadQueuesNonEmpty()` should resolve the issue.

In case when we request an address below the trim mark, the second call to `discoverAddressSpace(...)` will not get called because of:

```
        if (context.getGlobalPointer() >= maxAddress) {
            return areReadQueuesNonEmpty(context);
        }
```

Reply:
Turns out we should upgrade the global pointer after loading the checkpoint as well.